feat/manual-max-file-size-input: Add option to manually input max file size

krvspacetime opened the pull request:
## Description

Added a toggle to manually input the max file size instead of using the slider.

## Major changes:

- #### Files modified:
  
  - `git_form.jinja`
    
  - `index.py`
    
  - `process_query.py`
    
  - `utils.js`
    
- #### Added three form values / new params in the POST endpoint in `index.py`called:
  
  - `use_manual_input`: bool
    
  - `max_file_size_manual`: int
    
  - `size_unit:`str
    

    These are also added to the context in `process_query.py`

- #### Added three new inputs elements
  
  - `checkbox`: toggle use of slider vs. manual input
    
  - `number`: accepts manual file size in by manually inputing a value of using the increment/decrement triggers
    
  - `radio`: switch between accepted file sizes between`kb` or `mb`
    
- #### Keep new form/input states after receving results and submitting form
  
  - Added a function in `utils.js` called `restoreFormState` to keep current input/form values after a successfull POST request

## Minor changes

- Removed console log in `changePattern`function in `git_form.jinja`
  
- Reduced height of the absolute div (shadow effect) at large screens to 90% (`lg:h-90% sm:h-fulll)`) using media query in Tailwind to compensate to the slight change in dimesion of the div containg the form due to the addition of the new input element
  
- Removed unused variable `minp` in `utils.js`
  
- Added new context values in the home page to pass in default values called `default_file_size_manual`, `use_manual_input` and `size_unit`

cyclotruc commented:
Hi @krvspacetime  thank you for your contribution

I see some tests not passing

Regarding the changes, while I think it's a good idea to have the value editable with a text box, I think I want to keep the slider

I think the simplest way to keep both would be to make the value of the current slider editable by clicking it

what do you think?

krvspacetime commented:
Hello @cyclotruc, you're welcome and thank you for the feedback.

> I see some tests not passing

I think some of it is because I have to update the `test_flow_ingegration.py` and include the new form data values I added where necessary.

> Regarding the changes, while I think it's a good idea to have the value editable with a text box, I think I want to keep the slider
> 
> I think the simplest way to keep both would be to make the value of the current slider editable by clicking it

The slider is actually still there, it is set as the default way to input the max file size. The input I added only shows when toggled.

<img alt="Slider as default to input max_file_size" src="https://media-hosting.imagekit.io//0d609c3795854d0e/screencapture-localhost-8000-2025-03-19-11_39_18.png?Expires=1836963581&Key-Pair-Id=K2ZIVPTIP2VGHC&Signature=nPNY2IM~gTSNcXN-Mw0IwLoIxJQk9hl9DCAFLlWGDgO5YhRDTfTW2g5ryfCTzPzGL6jKvaa59cxjtL3zqZ-mnwGBlpfM3vN4BB4jBZWx54G7K8kMGwfjC-bNV8lG2kulXbVMJYPu7Zfwoz~eDCvBb7G05nCK5hwxRCwb7Ob1uK31xwuX8MAkEsYWVSUdli4u5Gvq8pHkbTVMtQqVtVNz~lofgEWffzBNLNBZofVNrWNdm6rwSWC3B8-QSyzf2-zTza~WH7A0udbymmiMTOkNZwiPle1hoXI0l-78L1a9zkyDQQLMz9HkquIqlTPWvSemAebxkAw-3IEc-sWeCJiTHg__" />
<img alt="Slider keeps state after results" src="https://media-hosting.imagekit.io//5352778e11784446/screencapture-localhost-8000-2025-03-19-11_41_28.png?Expires=1836963720&Key-Pair-Id=K2ZIVPTIP2VGHC&Signature=Q5bd4weY74VXSnn1-ty8uG3YgOexe7H3c5HPR5qrs09n8vIhWRMkMCfJ-BRYPrBI8iHGHj-BqDm6bNguhvstgkB44VHPKaF-p~RlyD629W3LSE8q~4OAC4QAeMRgT01F9CEVktZNK3K3EnQpohfELYAPUeq3kJY21m53SfTJm~VBUV5lxN8sdSEWOMVaKFsP9rKNdbzIBwJUXfDEkBtSGTg~zPpsQkL-B4LMu1EGCa4FEFCf8HMdky9S5YCqprd01ZVe1kkRT9zz55ggQUgH6wxvgvxYU07qo2gkzzhY1Xt8OhiXI~N1nPIc6bzHPasFyxX3XBSYcPzkyR5p6eUqSw__" />

> I think the simplest way to keep both would be to make the value of the current slider editable by clicking it

It's funny because this is actually what I tried initially. I simply replaced the span with an input that can be edited when clicked and then made a function to reverse the `log_slider_to_size` / `logSliderToSize` functions so the slider position can be calculated. Is that what you meant? If this is what you prefer, I can work on it.
